File: app/src/main/java/jp/juggler/subwaytooter/TabletColumnViewHolder.java
package jp.juggler.subwaytooter;

import android.support.v7.widget.RecyclerView;
import android.view.View;

import jp.juggler.subwaytooter.util.LogCategory;

class TabletColumnViewHolder extends RecyclerView.ViewHolder{
	static final LogCategory log = new LogCategory( "TabletColumnViewHolder" );
	
	
	final ColumnViewHolder vh;

	private int old_position = - 1;
	
	TabletColumnViewHolder(  ActMain activity, View v ){
		super( v );
		
		vh =new ColumnViewHolder( activity ,v);
		v.findViewById( R.id.vTabletDivider ).setVisibility( View.VISIBLE );
	}
	
	void bind(Column column,int position,int column_count ){
		log.d("bind. %d => %d ", old_position,position);
		old_position = position;

		vh.onPageDestroy( position );
		vh.onPageCreate( column, position,column_count );
		
		if( ! column.bFirstInitialized ){
			column.startLoading();
		}
	}
}

File: app/src/main/java/jp/juggler/subwaytooter/TabletColumnViewHolder.kt
package jp.juggler.subwaytooter

import android.support.v7.widget.RecyclerView
import android.view.View

import jp.juggler.util.LogCategory

internal class TabletColumnViewHolder(
	activity : ActMain,
	viewRoot : View
) : RecyclerView.ViewHolder(viewRoot) {
	
	companion object {
		val log = LogCategory("TabletColumnViewHolder")
	}
	
	val columnViewHolder : ColumnViewHolder
	
	private var pageIndex = - 1
	
	init {
		columnViewHolder = ColumnViewHolder(activity, viewRoot)
		// viewRoot.findViewById<View>(R.id.vTabletDivider).visibility = View.VISIBLE
	}
	
	fun bind(column : Column, pageIndex : Int, column_count : Int) {
		log.d("bind. %d => %d ", this.pageIndex, pageIndex)
		
		columnViewHolder.onPageDestroy(this.pageIndex)
		
		this.pageIndex = pageIndex
		
		columnViewHolder.onPageCreate(column, pageIndex, column_count)
		
		if(! column.bFirstInitialized) {
			column.startLoading()
		}
	}
	
	fun onViewRecycled() {
		log.d("onViewRecycled %d", pageIndex)
		columnViewHolder.onPageDestroy(pageIndex)
	}
}
